addons/website_mail/controllers/main.py

```diff
@@ -47,111 +46,90 @@ def _message_post_helper(res_model='', res_id=None, message='', token='', token_
     def website_message_subscribe(self, id=0, object=None, message_is_follower="on", email=False, **post):
         # TDE FIXME: check this method with new followers
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry['res.partner']
-        user_obj = request.registry['res.users']
-
-        _id = int(id)
-        _message_is_follower = message_is_follower == 'on'
-        _object = request.registry[object]
+        res_id = int(id)
+        is_follower = message_is_follower == 'on'
+        record = request.env[object].browse(res_id)
 
         # search partner_id
-        public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_ids = [user_obj.browse(cr, uid, uid, context).partner_id.id]
+        if request.env.user != request.website.user_id:
+            partner_ids = request.env.user.partner_id.ids
         else:
             # mail_thread method
-            partner_ids = _object._find_partner_from_emails(
-                cr, SUPERUSER_ID, [_id], [email], context=context, check_followers=True)
+            partner_ids = record.sudo()._find_partner_from_emails([email], check_followers=True)
             if not partner_ids or not partner_ids[0]:
                 name = email.split('@')[0]
-                partner_ids = [partner_obj.create(cr, SUPERUSER_ID, {'name': name, 'email': email}, context=context)]
-
+                partner_ids = request.env['res.partner'].sudo().create({'name': name, 'email': email}).ids
         # add or remove follower
-        if _message_is_follower:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
-            _object.message_unsubscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+        if is_follower:
+            record.check_access_rule('read')
+            record.sudo().message_unsubscribe(partner_ids)
             return False
         else:
-            _object.check_access_rule(cr, uid, [_id], 'read', context)
+            record.check_access_rule('read')
             # add partner to session
             request.session['partner_id'] = partner_ids[0]
-            _object.message_subscribe(cr, SUPERUSER_ID, [_id], partner_ids, context=context)
+            record.sudo().message_subscribe(partner_ids)
             return True
 
     @http.route(['/website_mail/is_follower'], type='json', auth="public", website=True)
     def call(self, model, id, **post):
-        id = int(id)
-        cr, uid, context = request.cr, request.uid, request.context
-
-        partner_obj = request.registry.get('res.partner')
-        users_obj = request.registry.get('res.users')
-        obj = request.registry.get(model)
-
-        partner_id = None
-        public_id = request.website.user_id.id
-        if uid != public_id:
-            partner_id = users_obj.browse(cr, SUPERUSER_ID, uid, context).partner_id
+        user = request.env.user
```

Comment:
I would rename it, because call is not the most explicit method name.

By the way, rename id to res_id.